app/src/main/java/ar/scacchipa/twittercloneapp/data/repository/CredentialRepository.kt

```diff
@@ -1,31 +1,56 @@
 package ar.scacchipa.twittercloneapp.data.repository
 
+import android.content.SharedPreferences
 import ar.scacchipa.twittercloneapp.data.IMapper
 import ar.scacchipa.twittercloneapp.data.datasource.IAuthDataSource
-import ar.scacchipa.twittercloneapp.data.model.UserAccessTokenData
+import ar.scacchipa.twittercloneapp.data.model.UserAccessToken
+import ar.scacchipa.twittercloneapp.domain.model.Credential
 import ar.scacchipa.twittercloneapp.domain.model.ResponseDomain
-import ar.scacchipa.twittercloneapp.domain.model.UserAccessTokenDomain
 import ar.scacchipa.twittercloneapp.utils.Constants
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
-class AuthorizationRepository(
-    private val genAccessTokenSource: IAuthDataSource,
-    private val mapper: IMapper<UserAccessTokenData, UserAccessTokenDomain>,
-    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
-): IAuthorizationRepository {
-    override suspend fun requestAccessToken(
+open class CredentialRepository(
+    private val credentialLocalSource: SharedPreferences,
+    private val accessTokenExternalSource: IAuthDataSource,
+    private val mapper: IMapper<UserAccessToken, Credential>,
+    private val dispatcherDefault: CoroutineDispatcher = Dispatchers.IO,
+    private val dispatcherIO: CoroutineDispatcher = Dispatchers.IO
+): ICredentialRepository {
+
+    override fun recoverLocalCredential(): Credential? {
```

Comment:
I think this should also be a suspend.

Reply:
All Repository layer function was converted to suspend function